go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -586,6 +541,182 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
 	return generated, nads, nil
 }
 
+// checkSubnetOverlaps validates that subnets from the current RA don't overlap
+// with any existing subnets in the same VRFs, both within the RA itself and
+// across different RAs
+func (c *Controller) checkSubnetOverlaps(ra *ratypes.RouteAdvertisements, selectedNetworks *selectedNetworks) error {
+	networkVRFsSubnets := createNetworkVRFsSubnetsMap(ra.Spec.TargetVRF, selectedNetworks)
+
+	// Check for overlaps within each VRF network
+	for vrf, subnets := range networkVRFsSubnets {
+		var err error
+		var overlappingSubnets []string
+		// Check for overlaps within current RA
+		if overlappingSubnets, err = checkSubnetSliceOverlaps(subnets); err != nil {
+			return fmt.Errorf("error checking subnet overlap within RouteAdvertisement %q in VRF %q: %w", ra.Name, vrf, err)
+		}
+		if len(overlappingSubnets) > 0 {
+			return fmt.Errorf("%w: overlapping CIDR detected within RouteAdvertisement %q in VRF %q: %v", errConfig, ra.Name, vrf, overlappingSubnets)
+		}
+		// Check for overlaps between current RA and other RAs in the same VRF
+		if err := c.checkCrossRAOverlaps(ra, vrf, subnets); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createNetworkVRFsSubnetsMap creates a map of VRF -> subnets from selectedNetworks
+func createNetworkVRFsSubnetsMap(targetVRF string, selectedNetworks *selectedNetworks) map[string][]string {
+	networkVRFsSubnets := make(map[string][]string)
+	if targetVRF == "auto" {
+		for networkVRF, network := range selectedNetworks.networkVRFs {
+			subnets := selectedNetworks.networkSubnets[network]
+			networkVRFsSubnets[networkVRF] = subnets
+		}
+	} else {
+		networkVRFsSubnets[targetVRF] = selectedNetworks.subnets
+	}
+	return networkVRFsSubnets
+}
+
+// checkSubnetSliceOverlaps checks for subnets overlap within a list of subnets
+func checkSubnetSliceOverlaps(subnets []string) ([]string, error) {
+	for i, subnet1 := range subnets {
+		for j, subnet2 := range subnets {
+			if i >= j {
+				continue
+			}
+			ipNets, err := util.ParseIPNets([]string{subnet1, subnet2})
+			if err != nil {
+				return nil, fmt.Errorf("failed to check subnet overlap between %s and %s: %w", subnet1, subnet2, err)
+			}
+			if len(util.IPNetOverlaps(ipNets[0], ipNets[1])) > 0 {
+				return []string{subnet1, subnet2}, nil
+			}
+		}
+	}
+	return []string{}, nil
+}
+
+// checkCrossRAOverlaps checks if subnets from current RA overlap with subnets from other RAs in the same VRF
+func (c *Controller) checkCrossRAOverlaps(ra *ratypes.RouteAdvertisements, targetVRF string, currentRASubnets []string) error {
+	allRAs, err := c.raLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
```

Comment:
We will need to do presubmit scale testing. Can you please reach out to Anil?

I would consider this candidate to caching in the controller as `map[vrf]map[ra][]*net.IPNet`

Reply:
Sure. Plainly speaking the unit test cover most cases (apart from the re-enqueue) so not a must but e2e is not a bad idea for the sake of future scale testing. 
Would you like it in a follow up PR or on this one?
Would this e2e test just be about basic conflict scenario (self+cross RA) , or something more complex?